pkg/controllermanager/apis/config/v1alpha1/defaults_test.go

```diff
@@ -29,245 +28,754 @@ import (
 
-			SetDefaults_SeedExtensionsCheckControllerConfiguration(obj)
-			Expect(obj.ConcurrentSyncs).To(PointTo(Equal(5)))
-			Expect(obj.SyncPeriod).To(PointTo(Equal(metav1.Duration{Duration: 30 * time.Second})))
+			Expect(obj.Controllers.ExposureClass).To(Equal(expected))
+		})
+
+		It("should not default fields that are set", func() {
+			obj := &ControllerManagerConfiguration{
+				Controllers: ControllerManagerControllerConfiguration{
+					ExposureClass: &ExposureClassControllerConfiguration{
+						ConcurrentSyncs: pointer.Int(10),
+					},
+				},
+			}
+			expected := obj.Controllers.ExposureClass.DeepCopy()
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
+
+			Expect(obj.Controllers.ExposureClass).To(Equal(expected))
 		})
 	})
 
-	Describe("#SetDefaults_SeedBackupBucketsCheckControllerConfiguration", func() {
-		It("should correctly default the SeedBackupBucketsCheck Controller configuration", func() {
-			obj := &SeedBackupBucketsCheckControllerConfiguration{}
+	Describe("QuotaControllerConfiguration defaulting", func() {
+		It("should default QuotaControllerConfiguration correctly", func() {
+			obj := &ControllerManagerConfiguration{}
+			expected := &QuotaControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(DefaultControllerConcurrentSyncs),
+			}
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
 
-			SetDefaults_SeedBackupBucketsCheckControllerConfiguration(obj)
-			Expect(obj.ConcurrentSyncs).To(PointTo(Equal(5)))
-			Expect(obj.SyncPeriod).To(PointTo(Equal(metav1.Duration{Duration: 30 * time.Second})))
+			Expect(obj.Controllers.Quota).To(Equal(expected))
+		})
+
+		It("should not default fields that are set", func() {
+			obj := &ControllerManagerConfiguration{
+				Controllers: ControllerManagerControllerConfiguration{
+					Quota: &QuotaControllerConfiguration{
+						ConcurrentSyncs: pointer.Int(10),
+					},
+				},
+			}
+			expected := obj.Controllers.Quota.DeepCopy()
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
+
+			Expect(obj.Controllers.Quota).To(Equal(expected))
+		})
+	})
+
+	Describe("SecretBindingControllerConfiguration defaulting", func() {
+		It("should default SecretBindingControllerConfiguration correctly", func() {
+			obj := &ControllerManagerConfiguration{}
+			expected := &SecretBindingControllerConfiguration{
+				ConcurrentSyncs: pointer.Int(DefaultControllerConcurrentSyncs),
+			}
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
+			Expect(obj.Controllers.SecretBinding).To(Equal(expected))
+		})
+		It("should not default fields that are set", func() {
+			obj := &ControllerManagerConfiguration{
+				Controllers: ControllerManagerControllerConfiguration{
+					SecretBinding: &SecretBindingControllerConfiguration{
+						ConcurrentSyncs: pointer.Int(10),
+					},
+				},
+			}
+			expected := obj.Controllers.SecretBinding.DeepCopy()
+			SetObjectDefaults_ControllerManagerConfiguration(obj)
+			Expect(obj.Controllers.SecretBinding).To(Equal(expected))
+		})
+	})
```

Comment:
```suggestion
	})
	
```
Similar changes for other occurrences as well.